aspnet/mvc/controllers/routing.rst

```diff
@@ -1,8 +1,140 @@
-.. include:: /../common/stub-topic.txt
+Routing to Controller Actions
+=============================
 
-|stub-icon| Routing to Controller Actions
-=========================================
+By `Rachel Appel`_
 
-.. include:: /../common/stub-notice.txt
+In this article:
 
-.. _issue: https://github.com/aspnet/Docs/issues/117
+.. contents:: Sections
+  :local:
+  :depth: 1
+
+Relationship to middleware
+--------------------------
+
+In the early days of the web you didn’t think about routing, as most web sites or applications simply referenced folders and files on a disk. If you had a URL for ``http://contoso.com/movies/GoneWithTheWind.htm``, then there was a folder called ``movies`` with a file called ``GoneWithTheWind.htm`` on the disk. With modern web frameworks like MVC you do not have to structure our content or application exactly how the world will see it. You can present it however you choose, and organize your software in the way that makes sense. `Routing <https://docs.asp.net/en/latest/fundamentals/routing.html>`_ is what you use to connect our code to the URL and features our application exposes. 
```

Comment:
cardinal rule - 1st occurrence is always ASP.NET Core MVC
Scottgu is very sensitive to us doing that. We did not invent MVC